pkg/porter/publish.go

```diff
@@ -327,29 +324,29 @@ func pushUpdatedImage(layout registry.Layout, origImg string, newImgName image.N
 // getNewImageNameFromBundleReference derives a new image.Name object from the provided original
 // image (string) using the provided bundleTag to clean registry/org/etc.
 func getNewImageNameFromBundleReference(origImg, bundleTag string) (image.Name, error) {
-	origName, err := image.NewName(origImg)
+	origImgRef, err := cnab.ParseOCIReference(origImg)
 	if err != nil {
-		return image.EmptyName, fmt.Errorf("unable to parse image %q into domain/path components: %w", origImg, err)
+		return image.EmptyName, err
 	}
 
-	bundleName, err := image.NewName(bundleTag)
+	bundleRef, err := cnab.ParseOCIReference(bundleTag)
 	if err != nil {
-		return image.EmptyName, fmt.Errorf("unable to parse bundle tag %q into domain/path components: %w", bundleTag, err)
+		return image.EmptyName, err
 	}
 
-	// Use the original image name with the bundle location to generate a randomized tag
-	source := path.Join(path.Dir(bundleName.Name()), origName.Name()) + ":" + bundleName.Tag()
-	nameHash := md5.Sum([]byte(source))
-	imgTag := hex.EncodeToString(nameHash[:])
+	tmpImage, err := cnab.CalculateTemporaryImageTag(origImgRef)
```

Comment:
Would this cause uniqueness issue when a different bundle tag is provided for the same original image? I remember I had to include the bundle tag information for the hash due to some uniqueness issue

Reply:
Right! I forgot that we changed to use digest for the referenced image. Then it shouldn't cause an issue anymore. Thank you for explaining it!

Reply:
I'm glad that you asked! I was a bit confused about why we were previously building up a composite string value based on both the source and the destination. 👍 Now we've got it written down somewhere. Let me add a quick comment to capture what we talked about here so we don't forget later.